src/Layers/xrRender/light_gi.cpp

```diff
@@ -21,8 +21,12 @@ void light::gi_generate()
         switch (flags.type)
         {
         case IRender_Light::POINT: dir.random_dir(random); break;
-        case IRender_Light::SPOT: dir.random_dir(direction, cone, random); break;
-        case IRender_Light::OMNIPART: dir.random_dir(direction, cone, random); break;
+        case IRender_Light::DIRECT:
```

Comment:
This changes behaviour! Please, keep pull request to it's topic.
This pull request says "optimize and refactor". Not adding new features. Please, remove new features and submit as separate pull request!